pinot-azure-filesystem/src/main/java/org/apache/pinot/filesystem/AzurePinotFS.java

```diff
@@ -100,6 +100,18 @@ public boolean delete(URI segmentUri, boolean forceDelete)
   @Override
   public boolean move(URI srcUri, URI dstUri, boolean overwrite)
       throws IOException {
+    if (exists(dstUri)) {
+      if (overwrite) {
+        delete(dstUri, true);
+      } else {
+        // dst file exists, returning
+        return false;
+      }
+    } else {
+      URI parentUri = Paths.get(dstUri).toUri();
```

Comment:
Am i reading this right that we are essentially creating dstUri as a directory but asking the user to pass the input param a a file?

Reply:
Correct the method according to the latest API doc we made in PinotFS interface.